json11.cpp

```diff
@@ -426,11 +419,14 @@ struct JsonParser final {
      */
     void consume_garbage() {
       consume_whitespace();
+      if (failed) return;
```

Comment:
'consume_whitespace()` does not trigger `failed`. why this check?
also a couple of lines below.